kani-compiler/src/codegen_cprover_gotoc/context/goto_ctx.rs

```diff
@@ -200,34 +203,58 @@ impl<'tcx> GotocCtx<'tcx> {
         self.symbol_table.lookup(name).unwrap()
     }
 
+    /// Ensures that a global variable `name` appears in the Symbol table and is initialized.
+    ///
+    /// This will add the symbol to the Symbol Table if not inserted yet.
+    /// This will register the initialization function if not initialized yet.
+    ///   - This case can happen for static variables, since they are declared first.
+    pub fn ensure_global_var_init<T, F>(
+        &mut self,
+        name: T,
+        is_file_local: bool,
+        is_const: bool,
+        t: Type,
+        loc: Location,
+        init: F,
+    ) -> &mut Symbol
+    where
+        T: Into<InternedString> + Clone + Debug,
+        F: Fn(&mut GotocCtx, Symbol) -> Expr,
+    {
+        let sym = self.ensure_global_var(name.clone(), is_file_local, t, loc);
+        sym.set_is_static_const(is_const);
+        if matches!(sym.value, SymbolValues::None) {
+            // Clone sym so we can use `&mut self`.
+            let sym = sym.clone();
+            let init_expr = SymbolValues::Expr(init(self, sym));
+            // Need to lookup again since symbol table might've changed.
+            let sym = self.symbol_table.lookup_mut(name).unwrap();
+            sym.value = init_expr;
+            sym
+        } else {
+            self.symbol_table.lookup_mut(name).unwrap()
```

Comment:
What if the symbol was present but its `init_expr` field was not defined, or defined to something different from the current `init` ?

Reply:
The former will enter the `if` block since `sym.value` will be `None`. We could add a sanity check for the value initialization to catch a potential rustc bug. That would require us to implement a proper expression comparison, which we don't have today and I believe it is out of scope of this PR.